### DISCUSSION TITLE
USB Removable Drives

### DISCUSSION THREAD

**Discussion by simonnzr** [tiny-pilot/tinypilot#1661] [Ideas]
Hi there,

I just got the Pro version of the tool and I'm happy with it.
Something I could really use would be a mountable writable USB drive, that I could access at least via ssh if not by the browser.  That way I could move files on/off the target PC, which I have no network attached to.

Is there anything in the works or any solution you can think of?

Cheers,
Simon

**Comment by mtlynch** [comment]
Thanks! This is something we're considering for the future.

There's no supported way to do it right now, but if you're interested in hacking around with the code, I believe you can enable read-write by changing this line in `/opt/tinypilot-privileged/scripts/mount-mass-storage`:


```bash
echo 1 > "${USB_FUNCTIONS_DIR_PATH}/lun.0/ro"
```

to this:


```bash
echo 0 > "${USB_FUNCTIONS_DIR_PATH}/lun.0/ro"
```

And then re-mounting a virtual media image.

Note that this is not currently supported, and the writes to the disk from either end are not guaranteed to be flushed to the underlying image. Updating TinyPilot to the next Pro release will overwrite any local changes to that file.

**Reply by simonnzr** [reply]
Great thanks this is work-around suits my needs in the meantime!
